examples/cuda/virtual_functions/example_dispatch.py

```diff
@@ -0,0 +1,500 @@
+        loads the function offset from the vtable, which differs by an 8-byte memory address offset.
+
+        Whereas the instruction for :code:`dynamic_foo_kernel` looks like::
+
+            LD.E R8, desc[UR4][R2.64] ;     # Load function offset from vtable
+
+        the instruction for :code:`dynamic_bar_kernel` looks like::
+
+            LD.E R8, desc[UR4][R2.64+0x8] ; # Load function offset from vtable
+        """
+        instructions_foo = decoder[self.DYNAMIC_KERNEL[MemberFunctionName.FOO]].instructions
+        instructions_bar = decoder[self.DYNAMIC_KERNEL[MemberFunctionName.BAR]].instructions
+
+        assert len(instructions_foo) == len(instructions_bar)
+
+        # There is exactly one instruction that differs.
+        [(instr_foo, instr_bar)] = [(instr_foo, instr_bar) for instr_foo, instr_bar in zip(instructions_foo, instructions_bar, strict=True) if instr_foo != instr_bar]
+
+        # The differing instruction is a load from generic memory.
+        matcher_load = LoadMatcher(arch=self.arch, readonly=False, memory=MemorySpace.GENERIC)
+        assert (matched_load_foo := matcher_load.match(instr_foo)) is not None
+        assert (matched_load_bar := matcher_load.match(instr_bar)) is not None
+
+        # The memory address differs by an 8-byte offset.
+        matcher_address = AddressMatcher(arch=self.arch, memory=MemorySpace.GENERIC)
+        assert (matched_address_load_foo := matcher_address.match(matched_load_foo.operands[1])) is not None
+        assert (matched_address_load_bar := matcher_address.match(matched_load_bar.operands[1])) is not None
+
+        assert matched_address_load_foo.reg == matched_address_load_bar.reg
+
+        assert matched_address_load_foo.offset is None
+        assert isinstance(matched_address_load_bar.offset, str)
+        assert matched_address_load_bar.offset == '0x8'
+
+class TestAllImplementationsInAllKernels(TestBinaryAnalysis):
+    """
+    The previous finding that the instructions generated for :code:`dynamic_foo_kernel` and :code:`dynamic_bar_kernel`
+    are identical except for the instruction that loads the function offset from the vtable is surprising, because it
+    indicates that the SASS codes for each kernel must each contain the implementations of both virtual functions, even
+    though each kernel only calls one of the virtual functions.
+
+    In fact, when inspecting the SASS code, we observe that the SASS code for each kernel contains the implementations
+    of all virtual functions of all derived classes in the compile unit, even those for which the compiler could be
+    expected to be able to deduce that they are not called.
+    """
+    DYNAMIC_KERNELS: tuple[str, ...] = (
+        f'{DispatchType.DYNAMIC}_{MemberFunctionName.FOO}_kernel',
+        f'{DispatchType.DYNAMIC}_{MemberFunctionName.BAR}_kernel',
+    )
+
+    def test_all_implementations_in_all_kernels(self, decoder: dict[str, Decoder]) -> None:
```

Comment:
```suggestion
    def test(self, decoder: dict[str, Decoder]) -> None:
```
(the class name is already explicit)